src/coreclr/src/vm/runtimehandles.cpp

```diff
@@ -1759,7 +1760,22 @@ void * QCALLTYPE RuntimeMethodHandle::GetFunctionPointer(MethodDesc * pMethod)
     // Ensure the method is active so
     // the function pointer can be used.
     pMethod->EnsureActive();
+
```

Comment:
This will skip the signature validation that is in `getCallInfo` right now. 

I think that we should move the signature validation from `getCallInfo` to right before the method is JITed (e.g. when we are setting the bit for the JIT; or where we are generating the stub code on x86) so that we are guaranteed to do it, no matter which path we get the function pointer through.

This will also make it pay for play (better for perf). The signature validation will be done only once we are actually going to execute the method.

Reply:
Good call. I will update those paths.

Reply:
Bah. The work I did to address this isn't complete. I need to rethink the current approach.

Reply:
Done.